test/jdk/javax/swing/JFileChooser/JFileChooserFontReset.java

```diff
@@ -66,9 +66,10 @@ public static void main(String args[]) throws Exception {
                     SwingUtilities.updateComponentTreeUI(fc);
                     Font curFont = fc.getFont();
                     System.out.println("current font " + curFont);
-                    if (curFont != null && !curFont.equals(origFont)) {
-                        throw new RuntimeException(
-                         "JFileChooser font did not reset after Look & Feel change");
+                    if ((curFont != null && !curFont.equals(origFont)) ||
+                        (origFont != null && !origFont.equals(curFont))) {
```

Comment:
The newly added condition is redundant because `origFont.equals(curFont) == curFont.equals(origFont)`. (It's possible `equals` returns different values but it violates its contract then.)

Reply:
I dont think it's redundant..There's a || check in between

If `curFont `is not null and `origFont `is null, then 1st condition `curFont != null && !curFont.equals(origFont)` is satisfied and it will throw exception

If `origFont `is not null and `curFont `is null, then 1st condition `curFont != null && !curFont.equals(origFont)` is not satisfied and 2nd condition `origFont != null && !origFont.equals(curFont)` is satisfied and it will throw exception. If we remove origFont.equals(curFont) check, it will not check the fix..

Reply:
`curFont != origFont && (curFont != null && !curFont.equals(origFont))`

Reply:
ok..modified but I guess mine was not "incorrect" either..

Reply:
> ok..modified but I guess mine was not "incorrect" either..

No, it was not wrong.

I believe this way is clearer though: if `curFont == origFont`, `equals` is bound to return `true`, so you can just avoid calling `equals` twice and make the condition shorter.